amanda::config via puppet scheme url now optional

mcgarrigle opened the pull request:
Hi,

The use-case for this PR is the requirement to maintain each amanda config file via a template instead of pulling the whole config tree from a puppet:// scheme URL that is maybe in a different module.

By setting ‘manage_configs_source’ to false, the module releases concern about the config file and allows the file resource to be defined by other means.

I believe this change to be non-breaking as existing behaviour is maintained by default, and may lay the groundwork for more comprehensive config management for the future.

Regards,
Pete.

bastelfreak commented on manifests/server.pp:
```diff
@@ -2,6 +2,7 @@
   $configs                  = [],
   $configs_directory        = undef,
   $manage_configs_directory = true,
+  $manage_configs_source    = true,
```
can you please add a datatype to this parameter?

bastelfreak commented on manifests/config.pp:
```diff
@@ -5,6 +5,7 @@
   $manage_configs_directory = true,
   $manage_dle               = false,
   $configs_source           = 'amanda/server',
+  $manage_configs_source    = true,
```
please add a datatype to the parameter

mcgarrigle replied:
Just new parameters? 
I have no problem with giving all the parameters a make-over.

mcgarrigle replied:
Sure thing. I am away from my build environment for about a another week, but will get right on it then..

mcgarrigle replied:
Is there anything else preventing this PR?

bastelfreak commented:
Hi @mcgarrigle, thanks for the PR! Can you please add a datatype to all new parameters?

mcgarrigle commented:
Hi,

I have updated my branch, and I have rebased against your master (I hope - I am not a git genius 😞).

I will open a new PR for the other changes discussed on this thread.

bastelfreak commented:
THanks @mcgarrigle !
